isthmus/src/main/java/io/substrait/isthmus/SqlExpressionToSubstrait.java

```diff
@@ -0,0 +1,165 @@
+
+  /**
+   * Converts the given SQL expression string to an {@link io.substrait.proto.ExtendedExpression }
+   *
+   * @param sqlExpression a SQL expression
+   * @param createStatements table creation statements defining fields referenced by the expression
+   * @return a {@link io.substrait.proto.ExtendedExpression }
+   * @throws SqlParseException
+   */
+  public ExtendedExpression convert(String sqlExpression, List<String> createStatements)
+      throws SqlParseException {
+    var result = registerCreateTablesForExtendedExpression(createStatements);
+    return executeInnerSQLExpression(
+        sqlExpression,
+        result.validator(),
+        result.catalogReader(),
+        result.nameToTypeMap(),
+        result.nameToNodeMap());
+  }
+
+  private ExtendedExpression executeInnerSQLExpression(
+      String sqlExpression,
+      SqlValidator validator,
+      CalciteCatalogReader catalogReader,
+      Map<String, RelDataType> nameToTypeMap,
+      Map<String, RexNode> nameToNodeMap)
+      throws SqlParseException {
+    RexNode rexNode =
+        sqlToRexNode(sqlExpression, validator, catalogReader, nameToTypeMap, nameToNodeMap);
+    NamedStruct namedStruct = toNamedStruct(nameToTypeMap);
+
+    ImmutableExpressionReference expressionReference =
+        ImmutableExpressionReference.builder()
+            .expression(rexNode.accept(this.rexConverter))
+            .addOutputNames("new-column")
+            .build();
+
+    List<io.substrait.extendedexpression.ExtendedExpression.ExpressionReference>
+        expressionReferences = new ArrayList<>();
+    expressionReferences.add(expressionReference);
+
+    ImmutableExtendedExpression.Builder extendedExpression =
+        ImmutableExtendedExpression.builder()
+            .referredExpressions(expressionReferences)
+            .baseSchema(namedStruct);
+
+    return new ExtendedExpressionProtoConverter().toProto(extendedExpression.build());
+  }
+
+  private RexNode sqlToRexNode(
+      String sql,
+      SqlValidator validator,
+      CalciteCatalogReader catalogReader,
+      Map<String, RelDataType> nameToTypeMap,
+      Map<String, RexNode> nameToNodeMap)
+      throws SqlParseException {
+    SqlParser parser = SqlParser.create(sql, parserConfig);
+    SqlNode sqlNode = parser.parseExpression();
+    SqlNode validSQLNode = validator.validateParameterizedExpression(sqlNode, nameToTypeMap);
+    SqlToRelConverter converter = createSqlToRelConverter(validator, catalogReader);
+    return converter.convertExpression(validSQLNode, nameToNodeMap);
+  }
+
+  SqlToRelConverter createSqlToRelConverter(
+      SqlValidator validator, CalciteCatalogReader catalogReader) {
+    return new SqlToRelConverter(
+        null,
+        validator,
+        catalogReader,
+        relOptCluster,
+        StandardConvertletTable.INSTANCE,
+        converterConfig);
+  }
+
+  private Result registerCreateTablesForExtendedExpression(List<String> tables)
```

Comment:
What would happen if we pass null here?

Reply:
Let me test that

Reply:
It may be possible, but an exception like this will be thrown: `org.apache.calcite.runtime.CalciteContextException: From line 1, column 1 to line 1, column 10: Unknown identifier 'L_ORDERKEY'`